resolvconf/resolvconf.go

```diff
@@ -4,21 +4,54 @@ package resolvconf
 import (
 	"bytes"
 	"io/ioutil"
+	"os"
 	"regexp"
 	"strings"
 	"sync"
 
 	"github.com/docker/docker/pkg/ioutils"
+	"github.com/docker/libnetwork/procfs"
 	"github.com/docker/libnetwork/resolvconf/dns"
 	"github.com/docker/libnetwork/types"
 	"github.com/sirupsen/logrus"
 )
 
 const (
-	// DefaultResolvConf points to the default file used for dns configuration on a linux machine
-	DefaultResolvConf = "/etc/resolv.conf"
+	defaultPath   = "/etc/resolv.conf"
+	alternatePath = "/run/systemd/resolve/resolv.conf"
 )
 
+var (
+	detectSystemdResolvConfOnce sync.Once
+	pathAfterSystemdDetection   = defaultPath
+)
+
+// Path return the path to the resolv.conf file that libnetwork should use.
+//
+// When systemd-resolved is running, the default /etc/resolv.conf points to
+// localhost. In this case, Path returns the alternative resolv.conf  at
+// /run/systemd/resolve/resolv.conf
+//
+// When attempting to determine if an alternate file is needed, if an error
+// is encountered, Path returns /etc/resolv.conf
+func Path() string {
+	detectSystemdResolvConfOnce.Do(func() {
+		pids, err := procfs.PidOf("systemd-resolved")
+		if err != nil {
+			logrus.Errorf("unable to check systemd-resolved status: %s", err)
+		}
+		if len(pids) > 0 && pids[0] > 0 {
+			_, err := os.Stat(alternatePath)
+			if err == nil {
+				logrus.Infof("systemd-resolved is running, so using resolvconf: %s", alternatePath)
```

Comment:
Probably we should use `systemd-resolved` config  only if `/etc/resolv.conf` points to a loopback address?

Reply:
e.g. when running with RootlessKit+slirp4netns/VPNKit, `systemd-resolved` would be visible (unless PIDNS is unshared) but `/etc/resolv.conf` would point to slirp4netns/VPNKit-builtin DNS.
Even in that case, using systemd DNS should be fine for almost all usecases, but not 100% sure.

Reply:
I'm not sure what the correct behavior is. After reading a bit about systemd-resolved i was surprised of the current behavior in moby, but this PR is about bringing whatever we have in moby to libnetwork in order to fix a difference between buildkit and non-buildkit builds.
If you think this won't work with rootlesskit, I suggest we fix it in a follow up.